crates/admin/protobuf/cluster_ctrl_svc.proto

```diff
@@ -85,11 +85,21 @@ message TrimLogRequest {
   uint64 trim_point = 2;
 }
 
-message CreatePartitionSnapshotRequest { uint32 partition_id = 1; }
+// Requests a partition snapshot to be archived. Optionally, a minimum
+// target LSN for the snapshot may be set and the log can be trimmed to the
+// archived LSN at the same time.
+message CreatePartitionSnapshotRequest {
+  uint32 partition_id = 1;
+  // Minimum LSN (inclusive) which must be covered by the snapshot
+  optional uint64 min_target_lsn = 2;
+  // Optionally, also trim the log to the archived LSN
+  optional bool trim_log = 3;
```

Comment:
If this is true, does that mean that the operation will fail if we couldn't trim all the way to the archived LSN? If no, then perhaps we should frame this more as a "hint" or at least express this in the comment.

Reply:
Might be useful if we can list/get the under going snapshot requests. with a possible progress indication and state.

Reply:
Yes, this can be a good future follow up